tabular/setup.py

```diff
@@ -34,7 +34,10 @@
         'lightgbm>=3.3,<3.4',
     ],
     'catboost': [
-        'catboost>=1.1,<1.2',
+        # CatBoost wheel build is not working correctly on darwin for CatBoost 1.2 on Python 3.8, so use old version in this case.
+        # https://github.com/autogluon/autogluon/pull/3190#issuecomment-1540599280
+        'catboost>=1.1,<1.2 ; python_version <= "3.8" and sys_platform == "darwin"',
```

Comment:
I just did some testing on all python 3.8 - 3.10, and all of them fail to install catboost 1.2.
Let's just not do it for macos no matter which python version it is

Reply:
Thanks for checking! I've updated the PR so that it should avoid installing 1.2 on MacOS, if you can give another try or we can run platform tests